examples/DictUpdateExamples.ipynb

```diff
@@ -0,0 +1,130 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aba8b4d5-199f-4fe0-97a2-5a2a3f43cbf2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "from traitlets import  Int, Dict, observe"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b11af66b-f256-4441-8c38-0283866024bd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class UpdateDictFromPythonWidget(ipyreact.ReactWidget):\n",
+    "    #note that when we add these traitlets, they will automatically be made available to the react component\n",
+    "    simple_dict = Dict({'foo':5, 'bar':8, 'baz':10}).tag(sync=True)\n",
+    "    count = Int(0).tag(sync=True)\n",
+    "\n",
+    "    @observe('count')\n",
+    "    def _observe_count(self, change):\n",
+    "        new_val = self.count\n",
+    "        sd = self.simple_dict.copy()\n",
```

Comment:
Maybe add a note saying that traitlets cannot detect mutations to a dict, and therefore we need to make a copy.